File: MicroSocialServer/MicroSocialServer/Schema/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MicroSocialServer.Schema
{
    class Message
    {
        public string messageBody;
        public DateTime time;
        public string senderName;
        public string recipientName;

        public string senderEmail;

        public Message(string messageBody, DateTime time, string senderName, string recipientName)
        {
            this.messageBody = messageBody;
            this.time = time;
            this.senderName = senderName;
            this.recipientName = recipientName;
        }

        public Message()
        {
        }
    }
}

File: MicroSocialAndroid/app/src/main/java/me/omegavesko/microsocial/android/alpha/schema/Message.java
package me.omegavesko.microsocial.android.alpha.schema;

import java.util.Date;

public class Message
{
    public String messageBody;
    public String time; // SQLite DateTime string, GSON doesn't understand it
    public String senderName;
    public String recipientName;
    public String senderEmail;

    public String displayName;

    public Message(String messageBody, String time, String senderName, String recipientName)
    {
        this.messageBody = messageBody;
        this.time = time;
        this.senderName = senderName;
        this.recipientName = recipientName;
    }

    public Message()
    {
    }
}